src/LaunchDarkly.Client/FeatureFlag.cs

```diff
@@ -0,0 +1,217 @@
+﻿using System.Collections.Generic;
+using LaunchDarkly.Client.Logging;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace LaunchDarkly.Client
+{
+    public class FeatureFlag
+    {
+        private static readonly ILog Logger = LogProvider.For<FeatureFlag>();
+
+        internal string Key { get; }
+        internal int Version { get; set; }
+        internal bool On { get; }
+        internal List<Prerequisite> Prerequisites { get; }
+        internal string Salt { get; }
+        internal List<Target> Targets { get; }
+        internal List<Rule> Rules { get; }
+        internal VariationOrRollout Fallthrough { get; }
+        internal int? OffVariation { get; }
+        internal List<JToken> Variations { get; }
+        internal bool Deleted { get; set; }
+
+        [JsonConstructor]
```

Comment:
Using the constructor pattern + annotation allows for immutability (notice most properties above don't have 'set;'